onnxmltools/convert/coreml/operator_converters/neural_network/InnerProduct.py

```diff
@@ -6,32 +6,41 @@
 
 from .....proto import onnx_proto
 from ....common._registration import register_converter
-
+from ....common._apply_operation import apply_reshape
 
 def convert_inner_product(scope, operator, container):
     params = operator.raw_operator.innerProduct
-    op_type = 'FC'
+    op_type = 'Gemm'
     inputs = [variable.full_name for variable in operator.inputs]
     outputs = [variable.full_name for variable in operator.outputs]
     attrs = {'name': operator.full_name}
 
-    name_w = scope.get_unique_variable_name(operator.full_name + '_W')
-    shape_w = [params.outputChannels, params.inputChannels]
-    inputs.append(name_w)
-    container.add_initializer(name_w, onnx_proto.TensorProto.FLOAT, shape_w, params.weights.floatValue)
+    shape_a = operator.inputs[0].type.shape
+    reshaped_tensor_name = scope.get_unique_variable_name('reshape_output')
+    apply_reshape(scope, operator.inputs[0].full_name, reshaped_tensor_name, container, desired_shape=[-1, params.inputChannels])
+    inputs[0] = reshaped_tensor_name
 
     name_b = scope.get_unique_variable_name(operator.full_name + '_B')
-    shape_b = [params.outputChannels]
+    shape_b = [params.outputChannels, params.inputChannels]
     inputs.append(name_b)
+    container.add_initializer(name_b, onnx_proto.TensorProto.FLOAT, shape_b, params.weights.floatValue)
+
+    name_c = scope.get_unique_variable_name(operator.full_name + '_C')
+    shape_c = [params.outputChannels]
+    inputs.append(name_c)
     if params.hasBias:
-        container.add_initializer(name_b, onnx_proto.TensorProto.FLOAT, shape_b, params.bias.floatValue)
+        container.add_initializer(name_c, onnx_proto.TensorProto.FLOAT, shape_c, params.bias.floatValue)
     else:
-        container.add_initializer(name_b, onnx_proto.TensorProto.FLOAT, shape_b, [0.] * shape_b[0])
+        container.add_initializer(name_b, onnx_proto.TensorProto.FLOAT, shape_c, [0.] * shape_b[0])
+    attrs['alpha'] = 1.0
+    attrs['beta'] = 1.0
+    attrs['transB'] = 1
```

Comment:
Do you need `attrs['transA'] = 0`? It's not marked as an optional [here](https://github.com/onnx/onnx/blob/master/docs/Operators.md#Gemm).